R/caseduplicate.R

```diff
@@ -7,14 +7,16 @@
 # (Miettinen/Schouten approach to directly estimating relative risks)
 estimate_duplicate <- function(formula, data, ...) {
   yvar <- as.character(all.vars(formula)[1])
-  data <- data %>%
-    dplyr::mutate(.clusterid = dplyr::row_number())
-  data <- dplyr::bind_rows(data,
-                           data %>%
-                             dplyr::rename(outc = dplyr::one_of(!!yvar)) %>%
-                             dplyr::filter(.data$outc == 1) %>%
-                             dplyr::mutate(outc = 0) %>%
-                             dplyr::rename(!!yvar := "outc"))
+
+  data <- data |>
+    dplyr::mutate(.clusterid = dplyr::row_number()) |>
+    dplyr::rename(outc = dplyr::one_of(!!yvar)) |>
+    tidyr::uncount(outc + 1) |>
```

Comment:
The see the elegance in this approach! However, in the spirit of my comment in #10, it would be nice *not* to rely on tidyverse functions down the road. Let us focus on testing and bug fixes for now.